pandas/core/arrays/base.py

```diff
@@ -610,3 +617,124 @@ def _ndarray_values(self):
         used for interacting with our indexers.
         """
         return np.array(self)
+
+
+class ExtensionScalarOpsMixin(object):
+    """
+    A base class for the mixins for different operators.
+    Can also be used to define an individual method for a specific
+    operator using the class method create_method()
+    """
+
+    @classmethod
+    def _create_method(cls, op, coerce_to_dtype=True):
+        """
+        A class method that returns a method that will correspond to an
+        operator for an ExtensionArray subclass, by dispatching to the
+        relevant operator defined on the individual elements of the
+        ExtensionArray.
+
+        Parameters
+        ----------
+        op:               An operator that takes arguments op(a, b)
+        coerce_to_dtype:  boolean indicating whether to attempt to convert
+                          the result to the underlying ExtensionArray dtype
+                          (default True)
+
+        Returns
+        -------
+        A method that can be bound to a method of a class
+
+        Example
+        -------
+        Given an ExtensionArray subclass called MyClass, use
+
+        >>> __add__ = ExtensionScalarOpsMixin.create_method(operator.add)
+
+        in the class definition of MyClass to create the operator
+        for addition.
+
+        """
+
+        op_name = ops._get_op_name(op, False)
+
+        def _binop(self, other):
+            def convert_values(parm):
```

Comment:
minor detail, but I find 'values' (or 'param') is clearer than 'parm'

Reply:
Changed